api/agent/func_logger.go

```diff
@@ -136,29 +103,35 @@ func newLineWriterWithBuffer(b *bytes.Buffer, w io.WriteCloser) io.WriteCloser {
 }
 
 func (li *lineWriter) Write(ogb []byte) (int, error) {
+	if atomic.LoadUint32(&li.closed) == 1 {
+		// we don't want to return 0/error or the container will shut down
+		return len(ogb), nil
+	}
 	li.b.Write(ogb) // bytes.Buffer is guaranteed, read it!
 
+	var n int
 	for {
-		b := li.b.Bytes()
-		i := bytes.IndexByte(b, '\n')
-		if i < 0 {
-			break // no more newlines in buffer
+		// read the line and advance buffer past it
+		l, err := li.b.ReadBytes('\n')
```

Comment:
unintended buffer consumption here if the buffer doesn't contain '\n', Close() will miss these bytes in that case.

Reply:
thanks, added test to cover this.